actions/npm-basic-checks/action.yml

```diff
@@ -0,0 +1,26 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of Node.
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: "Basic Checks and Tests"
+description: "Perform basic checks and tests for an NPM project"
+inputs:
+  node-version:
+    required: true
+    description: "the node version"
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ inputs.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+    - run: npm install
+      shell: bash
+    - run: npx prettier --check .
```

Comment:
hmmm we already have prettier job, also if anything we should run the version installed in the repo, not the latest from npx

Reply:
If you mean the composite action in this repo, that is for `yarn` projects, and also requires more configuration whereas I prefer to have a single action that has all the basic checks.

For npm projects, using `npx` is the recommended way:

https://prettier.io/docs/en/install.html

`npx` will use the locally-installed version of prettier if found, according to these docs.

Reply:
> If you mean the composite action in this repo, that is for `yarn` projects, and also requires more configuration whereas I prefer to have a single action that has all the basic checks.
> 
> For npm projects, using `npx` is the recommended way:
> 
> https://prettier.io/docs/en/install.html
> 
> `npx` will use the locally-installed version of prettier if found, according to these docs.

There's also a warning that you should not run npx if you don't have it added as a package. I suggest using `npm run prettier --check .` to ensure we're running the locally installed version and if it's not installed it should fail until we do.

> Note: If you forget to install Prettier first, npx will temporarily download the latest version. That’s not a good idea when using Prettier, because we change how code is formatted in each release! It’s important to have a locked down version of Prettier in your package.json. And it’s faster, too.